src/expansion/expand_variable.c

```diff
@@ -22,24 +22,34 @@ static char	*trim_single_quotes(t_expand_info *expand_info)
 	return (trim(expand_info));
 }
 
-char	*expand_variable(char *input, t_ctx *ctx)
+static const char	*construct_lookup(bool is_smart_expand)
```

Comment:
I think names like `determine_lookup`, `choose_lookup`, or `return_lookup` would be more appropriate.
In this project, the word "construct" is used for functions that allocate memory and fill in member variables.